app.go

```diff
@@ -379,8 +379,9 @@ type App struct {
 	errorHooks []ErrorHandler
 	validate   bool
 	// Used to signal shutdowns.
-	donesMu sync.RWMutex
-	dones   []chan os.Signal
+	donesMu     sync.Mutex
```

Comment:
I would add a comment here that this guards shutdownSig too.
```suggestion
	donesMu     sync.Mutex  // guards dones and shutdownSig
```